### PR TITLE
Vendor ratings cleanup

### PR DESCRIPTION
**Pull request by bhollis**
1. Fix ratings on non-unlocked collection items.
2. Add the colored star to ratings.
3. Fix ratings on definition-only vendors, like engram previews.
4. Some general code cleanup.

### CODE REVIEW

**Comment by bhollis** [comment]
5. Prevent waiting on item reviews to load before showing the move popup. Also, rate perks right away so that we show them once the reviews are loaded.

File: src/app/d2-vendors/VendorItemComponent.tsx [hunk 1]
```diff
@@ -106,19 +108,24 @@ export default class VendorItemComponent extends React.Component<Props, {}> {
...
+      let dimItem: DimItem | null = null;
 
       if (trackerService) {
         reviewData = trackerService.getD2ReviewDataCache().getRatingData(undefined, item.itemHash);
 
         if (reviewData && !reviewData.reviews) {
-          const reviewsData = await trackerService.getItemReviewAsync(item.itemHash);
-
-          Object.assign(reviewData, reviewsData);
+          trackerService.getItemReviewAsync(item.itemHash).then((reviewsData) => {
```
Comment (48klocs) [inline src/app/d2-vendors/VendorItemComponent.tsx]: As a takeaway for me, the trackerService should provide a simpler affordance to fetch rating data. This ugly service->cache->fetch flow is from an earlier iteration.

I tried to move this all into the service, but the data flows are kind of... entangled.

**Comment by 48klocs** [comment]
@bhollis lgtm

**Comment by duckmanBR** [comment]
The only problem is that you put the star on the wrong side of the number :sweat_smile:

**Pull request merged**